src/helper.cc

```diff
@@ -909,19 +911,10 @@ Helper::Client::sessionClosed(SessionBase &session)
 }
 
 void
-Helper::Session::HelperServerClosed(Session * const srv)
+Helper::SessionBase::HelperServerClosed(SessionBase * const srv)
 {
-    srv->parent->sessionClosed(*srv);
-    srv->dropQueued(*srv->parent);
-    delete srv;
-}
-
-// XXX: Essentially duplicates Helper::Session::HelperServerClosed() until we add SessionBase::helper().
-void
-helper_stateful_server::HelperServerClosed(helper_stateful_server *srv)
-{
-    srv->parent->sessionClosed(*srv);
-    srv->dropQueued(*srv->parent);
+    srv->helper().sessionClosed(*srv);
+    srv->dropQueued(srv->helper());
```

Comment:
We no longer need to pass the helper to dropQueued(), right? Please refactor to arrive at

```suggestion
    srv->dropQueued();
```